mlflow/utils/autologging_utils/safety.py

```diff
@@ -833,49 +963,73 @@ def _validate(autologging_call_input, user_call_input=None):
 
         if type(autologging_call_input) in [list, tuple]:
-            length_difference = len(autologging_call_input) - len(user_call_input)
-            assert (
-                length_difference >= 0
-            ), "{} expected inputs are missing from the call to the original function.".format(
-                length_difference
-            )
+            _assert_same_input_length(autologging_call_input, user_call_input)
             # If the autologging call input is longer than the user call input, we `zip_longest`
             # will pad the user call input with `None` values to ensure that the subsequent calls
             # to `_validate` identify new inputs added by the autologging call
             for a, u in itertools.zip_longest(autologging_call_input, user_call_input):
                 _validate(a, u)
         elif type(autologging_call_input) == dict:
-            assert set(user_call_input.keys()).issubset(set(autologging_call_input.keys())), (
-                "Keyword or dictionary arguments to original function omit"
-                " one or more expected keys: '{}'".format(
-                    set(user_call_input.keys()) - set(autologging_call_input.keys())
-                )
-            )
+            _assert_same_key_set(autologging_call_input, user_call_input)
             for key in autologging_call_input.keys():
                 _validate(autologging_call_input[key], user_call_input.get(key, None))
         else:
-            assert (
-                autologging_call_input is user_call_input
-                or autologging_call_input == user_call_input
-            ), (
-                "Input to original function does not match expected input."
-                " Original: '{}'. Expected: '{}'".format(autologging_call_input, user_call_input)
-            )
+            _assert_equivalence(autologging_call_input, user_call_input)
+
+    # Similar validation logic found in _validate, unraveling the list of arguments to exclude
+    # checks for any validation exempt positional arguments.
+    if user_call_args is None and autologging_call_args is not None:
+        _validate_new_input(autologging_call_args)
+    else:
+        _assert_same_type(autologging_call_args, user_call_args)
+        if autologging_call_args is None:
+            _assert_equivalence(autologging_call_args, user_call_args)
+        else:
+            _assert_same_input_length(autologging_call_args, user_call_args)
+            for index, autologging_call_arg, user_call_arg in itertools.zip_longest(
+                range(len(user_call_args)), autologging_call_args, user_call_args
+            ):
+                if not _is_arg_exempt_from_validation(
+                    autologging_integration,
+                    function_name,
+                    user_call_arg,
+                    positional_argument_index=index,
+                ):
+                    _validate(autologging_call_arg, user_call_arg)
```

Comment:
The set of positional arguments of a function is always a tuple. For simplicity, can we rewrite this logic with that assumption?